### PR TITLE
UserModel SetUp and Migrations

### PR DESCRIPTION
**Pull request by eriadhami**
@hassanhabib please check if I am in the right path

### CODE REVIEW

File: OtripleS.Web.Api/Brokers/Storage/StorageBroker.cs [hunk 1]
```diff
@@ -3,13 +3,14 @@
...
-using EFxceptions;
+using EFxceptions.Identity;
 using Microsoft.EntityFrameworkCore;
 using Microsoft.Extensions.Configuration;
+using OtripleS.Web.Api.Models.Users;
 
 namespace OtripleS.Web.Api.Brokers.Storage
 {
-    public partial class StorageBroker : EFxceptionsContext, IStorageBroker
+    public partial class StorageBroker : EFxceptionsIdentityContext<User>, IStorageBroker
```
Comment (hassanhabib) [inline OtripleS.Web.Api/Brokers/Storage/StorageBroker.cs:13]: You should probably do this as follows:
```chsarp
public partial class StorageBroker : EFxceptionsIdentityContext<User, Role, Guid>, IStorageBroker ...

```

Here's your Role Model: 

```csharp
using System;
using Microsoft.AspNetCore.Identity;

namespace Whatever ...
{
    public class Role : IdentityRole<Guid> { }
}

```

The Guid will just be for the unique identifier for the user

File: OtripleS.Web.Api/Models/Users/User.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+namespace OtripleS.Web.Api.Models.Users
+{
+    public class User : IdentityUser
+    {
+        public string Name { get; set; }
+        public string FamilyName { get; set; }
+        public UserStatus Status { get; set; }
+        public DateTimeOffset CreatedDate { get; set; }
+        public DateTimeOffset UpdatedDate { get; set; }
+    }
```
Comment (hassanhabib) [inline OtripleS.Web.Api/Models/Users/User.cs:18]: You can probably do this:
```csharp 
public class User : IdentityUser<Guid>
    {
        public override string UserName
        {
            get => base.Email;
            set => base.Email = value;
        }
    }

```
Along with the rest of the properties that you have.

Comment (eriadhami) [reply OtripleS.Web.Api/Models/Users/User.cs:36]: @hassanhabib resolved, thank you

File: OtripleS.Web.Api/Brokers/Storage/StorageBroker.cs (hunk 1 above)
Comment (eriadhami) [reply OtripleS.Web.Api/Brokers/Storage/StorageBroker.cs]: @hassanhabib resolved, many thanks

**Pull request merged**